android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncBroadcaster.kt

```diff
@@ -103,12 +103,26 @@ constructor(
   private fun Flow<PeriodicSyncJobStatus>.handlePeriodicSyncJobStatus(
     coroutineScope: CoroutineScope,
   ) {
-    this.onEach {
+    this.onEach { status ->
         syncListenerManager.onSyncListeners.forEach { onSyncListener ->
+          Timber.d("fhir sync worker...")
           onSyncListener.onSync(
-            if (it.lastSyncJobStatus != null) {
-              CurrentSyncJobStatus.Succeeded((it.lastSyncJobStatus as LastSyncJobStatus).timestamp)
-            } else it.currentSyncJobStatus,
+            if (status.lastSyncJobStatus != null) {
```

Comment:
Lets also update this existing check with the same condition. i.e.
```suggestion
            if (status.lastSyncJobStatus != null && status.lastSyncJobStatus is LastSyncJobStatus.Succeeded) {
```